editor/doc/http-api.md

```diff
@@ -0,0 +1,98 @@
+Editor HTTP API
+===============
+
+External tools can connect to the running editor and interact with it using a REST API.
+
+### Caveats
+* This HTTP API is currently in an experimental state. It might change drastically or even be removed completely if we deem it necessary. But hopefully it shouldn't come to that.
+* Currently, the port is assigned randomly from a broad subrange. The editor will log the local endpoint url if started from a terminal.
+
+### Overview
+The editor can be interacted with by performing GET requests to the following URLs:
+```
+http://localhost:[port]/command
+http://localhost:[port]/console
+```
+
+### The Command Endpoint
+The command endpoint exposes a subset of the commands available from the editor menu bar. To trigger a command, perform a GET request in the format:
+```
+http://localhost:[port]/command/[command]
+```
+You can obtain a list of the available commands along with a short description in JSON format from `http://localhost:[port]/command`. At the time of writing, this returns:
+```json
+{
+  "asset-portal"       : "Open the Asset Portal in a web browser.",
+  "async-reload"       : "Reload all modified files from disk.",
+  "build"              : "Build and run the project.",
+  "build-html5"        : "Build the project for HTML5 and open it in a web browser.",
+  "debugger-break"     : "Break into the debugger.",
+  "debugger-continue"  : "Resume execution in the debugger.",
+  "debugger-detach"    : "Detach the debugger from the running project.",
+  "debugger-start"     : "Start the project with the debugger attached.",
+  "debugger-step-into" : "Step into the current expression in the debugger.",
+  "debugger-step-out"  : "Step out of the current expression in the debugger.",
+  "debugger-step-over" : "Step over the current expression in the debugger.",
+  "debugger-stop"      : "Stop the debugger and the running project.",
+  "documentation"      : "Open the Defold documentation in a web browser.",
+  "donate-page"        : "Open the Donate to Defold page in a web browser.",
+  "editor-logs"        : "Show the directory containing the editor logs.",
+  "engine-profiler"    : "Open the Engine Profiler in a web browser.",
+  "fetch-libraries"    : "Download the latest version of the project library dependencies.",
```

Comment:
I discovered recently that UPnP is dangerous and unsafe. Just wandering if exposing "fetch libraries" could be dangerous? When someone clicks on "Fetch Libraries" they are looking at their added extensions. If it will be possible to trigger it externally then could someone use UPnP vulnerabilities to "silently" add a native code into an opened project? For indie devs that should not be a concern, but big companies might become a target since the source code is available for everyone.

Reply:
Hmm. On the second thought everyone will be using source control so I'm probably being paranoid :)